### PR TITLE
feat: auto-detect expression/function in js server

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/server/javascript.ts [hunk 1]
```diff
@@ -174,28 +173,25 @@ export async function evaluate(context: ExecutionContext, returnByValue: boolean
...
-      new Function('(' + functionText  + ')');
-    } catch (e2) {
-      // We tried hard to serialize, but there's a weird beast here.
-      throw new Error('Passed function is not well-serializable!');
+      new Function('(' + expression + ')');
+    } catch (e1) {
+      // This means we might have a function shorthand. Try another
+      // time prefixing 'function '.
+      if (expression.startsWith('async '))
+        expression = 'async function ' + expression.substring('async '.length);
```
Comment (yury-s) [inline src/server/javascript.ts:186]: I'd move this to the js client where we now if it's a function or not and just need to update the serialization logic to replace the prefix if needed. Could be done separately.

**Review by yury-s — APPROVED** [review]
Looks good as long as Dmitry is cool with removing the source maps support.

File: src/server/webkit/protocol.ts [hunk 2]
```diff
@@ -7428,7 +7428,7 @@ the top of the viewport and Y increases as it proceeds towards the bottom of the
       /**
        * Indicates if this type description has been type Function.
        */
-      isFunction: boolean;
+      isFunction: boolean | undefined;
```
Comment (dgozman) [inline src/server/webkit/protocol.ts:7431]: How come our protocol.ts is changed without browsers.json? It should be rolled together with the browser.

**Review by dgozman — APPROVED** [review]

File: src/server/webkit/protocol.ts (hunk 2 above)
Comment (pavelfeldman) [reply src/server/webkit/protocol.ts:7431]: oops, this is just a mass replace

File: src/server/javascript.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/server/javascript.ts:186]: I need to patch 10 call sites for that :/

**Pull request merged**